coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala

```diff
@@ -0,0 +1,73 @@
+package filodb.coordinator.queryplanner
+
+import filodb.core.query.QueryContext
+import filodb.query.{BinaryJoin, LabelValues, LogicalPlan, SeriesKeysByFilters, SetOperator}
+import filodb.query.exec._
+
+/**
+  * SinglePartitionPlanner is responsible for planning in situations where time series data is
+  * distributed across multiple clusters.
+  *
+  * @param planners map of clusters names in the local partition to their Planner objects
+  * @param plannerSelector a function that selects the planner name given the metric name
+  *
+  */
+class SinglePartitionPlanner(planners: Map[String, QueryPlanner], plannerSelector: String => String,
+                             datasetMetricColumn: String)
+  extends QueryPlanner {
+
+  def materialize(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
+
+    logicalPlan match {
+      case lp: BinaryJoin          => materializeBinaryJoin(lp, qContext)
+      case lp: LabelValues         => materializeLabelValues(lp, qContext)
+      case lp: SeriesKeysByFilters => materializeSeriesKeysFilters(lp, qContext)
+      case _                       => materializeSimpleQuery(logicalPlan, qContext)
+
+    }
+  }
+
+  private def materializeSimpleQuery(logicalPlan: LogicalPlan, qContext: QueryContext): ExecPlan = {
+
+    planners.get(plannerSelector(LogicalPlanUtils.getMetricName(logicalPlan).head)).get.
```

Comment:
What if there is no metric name? Simple scalar queries would throw an exception

Reply:
Also,  @vishramachandran what do you think about moving this logic to QS itself. Basically planSelector will accept logicalPlan and let that function decide which planner has to be picked up.

Reply:
That capability is owned by query engine. 
It is a simple fix.
In the case where there is no metric name, any one of the planners can be used.

Reply:
Thats why i suggested that instead of QueryEngine to decide which plan to use as default, 
QS will give you the default planner.

Reply:
To summarize:

1. remove BadQueryException from getMetricName
2. if metricName is None or Nil, use planners.head  to get the top planner in the map and route it to that guy

Reply:
Ohh yes. I missed this. Thanks for letting me know! I have made the changes and also added test case for Scalar query